packages/samples/headless-react/src/components/facet/facet-hook.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  buildFacet,
+  Controller,
+  Facet,
+  FacetProps,
+  FacetState,
+  SearchEngine,
+} from '@coveo/headless';
+import React, {useEffect, useState} from 'react';
+
+/**
+ * Example of a specific hook for function component
+ * We would need to create one of those for each Headless controller, and for each sub package of headless
+ * For example, `useFacet` from '@coveo/headless/react', then `useFacet` from '@coveo/headless/productlisting/react' etc.
+ *
+ */
+export const useFacet = (
+  facet: Facet
```

Comment:
Would it be possible for `useFacet` to take `FacetProps` instead, and be in charge of creating and returning the controller?

> I wasn't able to come up with a pattern where we could actually create the controller for our users inside the hook, and that would "only run once"

Did you specify a facet id?

Otherwise, I believe the `useContext` can retrieve the engine, but would the returned state need to become an object to trigger updates correctly?

```
{
   controller, // does not change
   controllerState // changes
}
```